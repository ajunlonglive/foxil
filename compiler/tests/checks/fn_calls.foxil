; funcs calls checks
func @func.args.0() void {
}

func @func.args.2(i32 %f) void {
}

func @func.args.3(i32 %f, i32 %fa, i32 %m) void {
}

func @main(i32 %argc, str* %argv) i32 {
    call void @func.args.0(i32 5)
    call void @func.args.2(i32 5, i32 5)
    call void @func.args.3(i32 5, i32 5)
    ret i32 0
}
