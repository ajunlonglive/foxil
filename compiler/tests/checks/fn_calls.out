compiler/tests/checks/fn_calls.foxil:12:4: error: too many arguments to function ‘@func.args.0’ (0 argument(s) are expected, not 1)
    12 |     call void @func.args.0(i32 5)
       |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
compiler/tests/checks/fn_calls.foxil:13:4: error: too many arguments to function ‘@func.args.2’ (1 argument(s) are expected, not 2)
    13 |     call void @func.args.2(i32 5, i32 5)
       |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
compiler/tests/checks/fn_calls.foxil:14:4: error: too few arguments to function ‘@func.args.3’ (3 argument(s) are expected, not 2)
    14 |     call void @func.args.3(i32 5, i32 5)
       |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
